src/isomorphic/modern/class/ReactNoopUpdateQueue.js

```diff
@@ -11,21 +11,35 @@
 
 if (__DEV__) {
   var warning = require('fbjs/lib/warning');
+  var errorInfo = {};
 }
 
 function warnNoop(publicInstance, callerName) {
   if (__DEV__) {
     var constructor = publicInstance.constructor;
+    const currentComponent =
+      (constructor && (constructor.displayName || constructor.name)) ||
+      'ReactClass';
+    const currentComponentError =
```

Comment:
It feels odd to duplicate the message here. Can we just use the `callerName` and component name combined as a key, without the warning itself?